packages/analytics-browser/src/browser-client.ts

```diff
@@ -130,6 +133,22 @@ export class AmplitudeBrowser extends AmplitudeCore implements BrowserClient {
     connector.eventBridge.setEventReceiver((event) => {
       void this.track(event.eventType, event.eventProperties);
     });
+
+    // Step 8: Add network listeners for offline mode
```

Comment:
I would prefer if this is done in the Plugin instead. Plugins should  have access to `client.flush()` via `plugin.setup(config, client)`

Reply:
I moved all the offline logic to the plugin at https://github.com/amplitude/Amplitude-TypeScript/pull/644/commits/8d14bfc77bb7a7c98fd82309b9cd9a4e990f80bd#diff-0c1d0fcf8156764bdf906a8f23ee6243eff9afeda0799e3551ad83237562e4f0